### PR TITLE
CLI: introduce a non-strict mode to all tools

### PR DESCRIPTION
**Pull request by andlaus**
This is a spin-off of #256: it adds a non-strict mode to all comand line tools whilst the patch that used to be in #256 only dealt with `list` and `compare`. using this mode is more robust regarding incorrect input files, bugs and unimplemented features in odxtools, but it might cause undefined behaviour.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

**Comment by kayoub5** [comment]
suggestion: use 
BooleanOptionalAction
see https://stackoverflow.com/a/68837302/1438522

requires Python 3.9, but we can safely drop support for 3.8 since it reaches end of life October 2024, see https://endoflife.date/python

**Comment by andlaus** [comment]
I'd like to be a bit more conservative here and wait until Ubuntu 20.04 LTS reaches EOL. that should be April/May 2025...

File: odxtools/cli/main.py [hunk 1]
```diff
@@ -50,4 +60,12 @@ def start_cli() -> None:
 
     for tool in tool_modules:
         if tool._odxtools_tool_name_ == args.subparser_name:
-            tool.run(args)
+            orig_strict = odxtools.exceptions.strict_mode
+            odxtools.exceptions.strict_mode = not args.no_strict
+            try:
+                tool.run(args)
+            except:
+                raise
```
Comment (kayoub5) [inline odxtools/cli/main.py:68]: try/finally enough, the except block is not required here

Comment (andlaus) [reply odxtools/cli/main.py:68]: that's true if no backtrace ought to be printed if an exception happens to seep out of the tool's `run()` function and if `start_cli()` is guaranteed to never be called from other python code. I think that of these two assertions,  only the latter applies. (I would prefer to be have a fighting chance of finding bugs based on the output of the CLI tools...)

Comment (andlaus) [reply odxtools/cli/main.py:68]: (that's due to the fact that so far, no non-standalone callers `start_cli()` exist: `test_cli.py` calls the `run()` functions of the tools directly.)

Comment (kayoub5) [reply odxtools/cli/main.py:68]: I meant the code should look like this

```python
            try:
                tool.run(args)
            finally:
                odxtools.exceptions.strict_mode = orig_strict
```

Comment (andlaus) [reply odxtools/cli/main.py]: okay, my mistake: I somehow assumed that this way exceptions thrown by `tool.run()` get silently ignored whilst in reality they get passed through. (I don't know what made me think that, probably a brain fart :/. anyway: fixed.)

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**